engine/src/vega_cast_utils.h

```diff
@@ -91,20 +91,17 @@ inline long locale_aware_stol(const std::string& s, const long default_value = 0
 }
 
 template<class TargetType, class SourceType>
-inline TargetType* vega_dynamic_cast_ptr(SourceType* from) {
+inline TargetType* vega_dynamic_cast_ptr(SourceType* from, const bool allow_null = false) {
     TargetType* ret_val = nullptr;
     try {
         ret_val = dynamic_cast<TargetType*>(from);
     } catch (std::bad_cast& e) {
         VS_LOG_AND_FLUSH(fatal, (boost::format("Fatal Error '%1%' casting type %2%* to %3%*") % e.what() % typeid(SourceType).name() % typeid(TargetType).name()));
         VSExit(-422);
-    } catch (std::exception& e) {
-        VS_LOG_AND_FLUSH(fatal, (boost::format("Fatal Error '%1%' casting type %2%* to %3%*") % e.what() % typeid(SourceType).name() % typeid(TargetType).name()));
-        VSExit(-423);
```

Comment:
CLion informs me that this block is unreachable code. So I guess `std::bad_cast` is the only type of exception that can be thrown here?

It certainly cleans the code up nicely to take this out.